core/ledger/kvledger/txmgmt/statedb/statecouchdb/statecouchdb_test.go

```diff
@@ -723,27 +723,92 @@ func TestHandleChaincodeDeployErroneousIndexFile(t *testing.T) {
+
+	// though the indexColorSortName.json is processed before indexSizeSortName.json as per the order,
+	// the later would replace the former as the index names are the same. Hence, a query using the color
+	// field in sort should fail.
+	queryString = `{"selector":{"owner":"fred"}, "sort": [{"color": "desc"}]}`
+	queryUsingIndex = func() bool {
+		_, err = db.ExecuteQuery("ns1", queryString)
+		if err != nil {
+			return false
+		}
+		return true
+	}
+	assert.Never(t, queryUsingIndex, 2*time.Second, 100*time.Millisecond, "error should have occurred as there is no index on color field")
+
 	indexData = map[string][]byte{
-		"META-INF/statedb/couchdb/indexes/badSyntax.json": []byte(badSyntaxFileContent),
+		"META-INF/statedb/couchdb/indexes/d-indexColorSortName.json":                           []byte(`{"index":{"fields":[{"color":"desc"}]},"ddoc":"indexSizeSortName","name":"indexSizeSortName","type":"json"}`),
+		"META-INF/statedb/couchdb/indexes/c-indexSizeSortName.json":                            []byte(`{"index":{"fields":[{"size":"desc"}]},"ddoc":"indexSizeSortName","name":"indexSizeSortName","type":"json"}`),
+		"META-INF/statedb/couchdb/indexes/badSyntax.json":                                      []byte(badSyntaxFileContent),
+		"META-INF/statedb/couchdb/collections/collectionMarbles/indexes/indexCollMarbles.json": []byte(`{"index":{"fields":["docType","owner"]},"ddoc":"indexCollectionMarbles", "name":"indexCollectionMarbles","type":"json"}`),
 	}
-	assert.Error(t, indexCapable.ProcessIndexesForChaincodeDeploy("ns1", indexData))
 
-	queryString := `{"selector":{"owner":"fred"}, "sort": [{"size": "desc"}]}`
-	queryUsingIndex := func() bool {
+	// as the indexes are sorted by file names, the order of index processing would be
+	// (1) indexCollMarbles.json, (2) badSyntax.json, (3) c-indexSizeSortName, (4) d-indexColorSortName.
+	// As the c-indexSizeSortName.json and d-indexColorSortName has the same index name but different
+	// index fields, the later would replace the former, i.e., index would be created on color field
+	// rather than the size field. Further, the index with bad syntax would not stop the processing
+	// of other valid indexes.
+	assert.NoError(t, indexCapable.ProcessIndexesForChaincodeDeploy("ns1", indexData))
+
+	// due to the change in order, the query using the size index would fail
+	queryString = `{"selector":{"owner":"fred"}, "sort": [{"size": "desc"}]}`
+	queryUsingIndex = func() bool {
+		_, err = db.ExecuteQuery("ns1", queryString)
+		if err != nil {
+			return false
+		}
+		return true
+	}
+	assert.Never(t, queryUsingIndex, 2*time.Second, 100*time.Millisecond, "error should have occurred as there is no index on color field")
+
+	// due to the change in order, the query using the color index would pass
+	queryString = `{"selector":{"owner":"fred"}, "sort": [{"color": "desc"}]}`
+	queryUsingIndex = func() bool {
```

Comment:
This pretty much seems the repeat of what is already tested in the previous block. Did I miss anything?

Reply:
True. Removed it.